pkg/models/change_table_record.go

```diff
@@ -12,10 +12,39 @@ import (
 type ChangeTableRecord struct {
 	GUID        string                      `json:"guid"`
 	TableID     int                         `json:"table_id"`
+	TableName   string                      `json:"table_name"`
 	PrimaryKey  uuid.UUID                   `json:"primary_key"`
 	ForeignKey  *uuid.UUID                  `json:"foreign_key"`
 	Action      string                      `json:"action"`
-	Fields      map[string]interface{}      `json:"fields"`
+	Fields      ChangedFields               `json:"fields"`
 	ModifiedDts *time.Time                  `json:"modified_dts"`
 	ModifiedBy  *authentication.UserAccount `json:"modified_by"`
 }
+
+// ChangedFieldValue represents an interface for changed field values.
+type ChangedFieldValue interface {
+	IsChangedFieldValue()
+}
+
+// ChangedFields contains a slice of changed fields.
+type ChangedFields struct {
+	Changes []*Field `json:"changes"`
+}
+
+// IsChangedFieldValue is a method to satisfy the ChangedFieldValue interface for ChangedFields.
+func (ChangedFields) IsChangedFieldValue() {}
+
+// Field represents an individual field, including its name and associated value.
+type Field struct {
+	Name  string            `json:"name"`
+	Value ChangedFieldValue `json:"value"`
+}
+
+// FieldValue represents the new and old values of a changed field.
+type FieldValue struct {
+	New *string `json:"new"`
```

Comment:
same thing here, this might make more sense as an interface type. Thoughts?

Reply:
Same as the other comment